NickvisionCavalier.GNOME/Controls/AddProfileDialog.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using static NickvisionCavalier.Shared.Helpers.Gettext;
+
+namespace NickvisionCavalier.GNOME.Controls;
+
+/// <summary>
+/// A dialog for showing a message
+/// </summary>
+public class AddProfileDialog
+{
+    private readonly Adw.MessageDialog _dialog;
+
+    public string ProfileName;
+    public event EventHandler<MessageDialogResponse>? OnDialogClosed;
```

Comment:
Also, why is this extra event needed? You can just bind to AddProfileDialog.OnResponse += which will occur when the dialog gets a response...(the event on line 48)